enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,469 @@
+cluster profile will configure these components as appropriate
+for a single node.
+
+When we are deploying a cluster with the `single-node-production-edge`
+cluster profile, the relevant operators should support a non-HA
+configuration that makes the correct adjustments to the deployment
+(e.g., `cluster-ingress-operator` should deploy a single router,
+`cluster-etcd-operator` should deploy the `etcd-member` [without
+waiting for 3 master
+nodes](https://github.com/openshift/cluster-etcd-operator/blob/98590e6ecfe282735c4eff01432ae40b29f81202/pkg/etcdenvvar/etcd_env.go#L72))
+
+In addition, some components are not relevant for this cluster profile
+(e.g. console, cluster-autoscaler, keepalived for ingressVIP and
+apiVIP) and shouldn't be deployed at all.
+
+#### cluster-etcd-operator
+
+By default, `cluster-etcd-operator` will not deploy the etcd cluster
+without minimum of 3 master nodes. This can be changed by enabling
+`useUnsupportedUnsafeNonHANonProductionUnstableEtcd`.
+
+```shell
+# allow etcd-operator to start the etcd cluster without minimum of 3 master nodes
+oc patch etcd cluster --type=merge -p="$(cat <<- EOF
+
+ spec:
+   unsupportedConfigOverrides:
+     useUnsupportedUnsafeNonHANonProductionUnstableEtcd: true
+EOF
+)"
+```
+
+Even with the unsupported feature flag, `etcd-quorum-guard` still
+requires 3 nodes due to its replica count. The `etcd-quorum-guard`
+Deployment is managed by the `cluster-version-operator`, so it needs
+to be marked as unmanaged before it can be scaled down.
+
+```shell
+# tell the cluster-version-operator not to manage etcd-quorum-guard
+oc patch clusterversion/version --type='merge' -p "$(cat <<- EOF
+ spec:
+    overrides:
+      - group: apps/v1
+        kind: Deployment
+        name: etcd-quorum-guard
+        namespace: openshift-machine-config-operator
+        unmanaged: true
+EOF
+)"
+
+# scale down etcd-quorum-guard
+oc scale --replicas=1 deployment/etcd-quorum-guard -n openshift-etcd
+```
+
+#### cluster-authentication-operator
+
+By default, the `cluster-authentication-operator` will not deploy
+`OAuthServer` without minimum of 3 master nodes. This can be change by
+enabling `useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer`.
+
+```shell
+# allow cluster-authentication-operator to deploy OAuthServer without minimum of 3 master nodes
+oc patch authentications.operator.openshift.io cluster --type=merge -p="$(cat <<- EOF
+
+ spec:
+   managementState: "Managed"
+   unsupportedConfigOverrides:
+     useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer: true
+EOF
+)"
+```
+
+#### cluster-ingress-operator
+
+By default, the `cluster-ingress-operator` deploys the router with 2
+replicas. On a single node one will fail to start and the ingress will
+show as degraded.
+
+```shell
+# patch ingress operator to run a single router pod
+oc patch -n openshift-ingress-operator ingresscontroller/default --type=merge --patch '{"spec":{"replicas": 1}}'
+```
+
+#### machine-config-operator
```

Comment:
Are we expecting any day2 configuration change using MCO on the single-node cluster in its lifetime?

Reply:
yep. Applying rt kernel, or other telco related configurations (performance tunings)

Reply:
+1 to sinny's question as this is super impt.. there's an open question (#3 below) that mentions this but it feels like this needs a lot more consideration & consensus followed by memorialising that in the enhancement here (and probably updating the risks section as well) as opposed to just leaving it as a tbd question.

Reply:
Applying any configuration (like rt kernel, kargs, files, etc) during install time shouldn't have much risk on single node. For any day2 changes MCO is going to drain all running workload and then reboot(few exception). For a production level node, we need to discuss where running workload will be drained? If something goes wrong in between, there is no way to get  must-gather to perform troubleshooting. I am sure there will be more issues which may come into picture as MCO has been designed keeping in mind that it will run on multi node cluster (with minimum 3 compute and control plane).